contracts/pangolin-periphery/PangolinBridgeMigrationRouter.sol

```diff
@@ -0,0 +1,358 @@
+pragma solidity ^0.7.6;
+
+import "../pangolin-core/interfaces/IPangolinERC20.sol";
+import "../pangolin-lib/libraries/TransferHelper.sol";
+import "./interfaces/IBridgeToken.sol";
+import "./libraries/Roles.sol";
+import "./libraries/PangolinLibrary.sol";
+import "hardhat/console.sol";
+
+contract PangolinBridgeMigrationRouter {
+    using SafeMath for uint;
+    using Roles for Roles.Role;
+
+    Roles.Role private adminRole;
+    mapping(address => address) public bridgeMigrator;
+
+    constructor() public {
+        adminRole.add(msg.sender);
+    }
+
+    // safety measure to prevent clear front-running by delayed block
+    modifier ensure(uint deadline) {
+        require(deadline >= block.timestamp, 'PangolinBridgeMigrationRouter: EXPIRED');
+        _;
+    }
+
+    // makes sure the admin is the one calling protected methods
+    modifier onlyAdmin() {
+        require(adminRole.has(msg.sender), 'PangolinBridgeMigrationRouter: Unauthorized');
+        _;
+    }
+
+    /**
+     * @notice Given an address, this address is added to the list of admin.
+     * @dev Any admin can add or remove other admins, careful.
+     * @param account The address of the account.
+     */
+    function addAdmin(address account) external onlyAdmin {
+        adminRole.add(account);
+    }
+
+    /**
+     * @notice Given an address, this address is added to the list of admin.
+     * @dev Any admin can add or remove other admins, careful.
+     * @param account The address of the account.
+     */
+    function removeAdmin(address account) external onlyAdmin {
+        adminRole.remove(account);
+    }
```

Comment:
should not allow to remove admin if there's only 1 admin left

Reply:
true

Reply:
Would it be ok if I add a validation to not allow an admin to remove itself? It would yield the same result but with a little bit more constraint

Reply:
the goal is to make the contract simple, not overly complex which brings more trust to it

Reply:
Fixed, resolving, feel free to reopen